components/alert/alert-toast.js

```diff
@@ -256,19 +332,37 @@ class AlertToast extends LitElement {
 				}
 			}
 			this.setAttribute('role', 'alert');
+			await this.updateComplete;
+			const height = this._innerContainer.offsetHeight;
+			this.dispatchEvent(new CustomEvent(
+				'd2l-alert-toast-open', {
+					bubbles: true,
+					composed: false,
+					detail: { height }
+				}
+			));
```

Comment:
This is going to force a browser repaint. Since you don't need to fire the event synchronously anyway, it might be slightly more performant to put this inside an rAF? Then `_openChanged` doesn't need to be `async` anymore and I don't think you need to wait for `updateComplete`.

```suggestion
			requestAnimationFrame(() => {
				const height = this._innerContainer.offsetHeight;
				this.dispatchEvent(new CustomEvent(
					'd2l-alert-toast-open', {
						bubbles: true,
						composed: false,
						detail: { height }
					}
				));
			});
```

If that works, the same could be applied to the `else` block.

Reply:
It'll work yeah, but any time you access `offsetHeight` it forces the browser to paint, so it's a good general practice to wrap those reads inside a `rAF` to minimize the repaints.